yarn-project/end-to-end/src/simulators/token_simulator.ts

```diff
@@ -3,8 +3,8 @@ import { type AztecAddress, type DebugLogger } from '@aztec/aztec.js';
 import { type TokenContract } from '@aztec/noir-contracts.js/Token';
 
 export class TokenSimulator {
-  private balancesPrivate: Map<AztecAddress, bigint> = new Map();
-  private balancePublic: Map<AztecAddress, bigint> = new Map();
+  private balancesPrivate: Map<string, bigint> = new Map();
+  private balancePublic: Map<string, bigint> = new Map();
```

Comment:
Why did you have to change the types here?

Reply:
The `AztecAddress<Bob>` from `accounts` and `AztecAddress<Bob>` from `wallets` would be two different keys, so `Bob` ended up having "two balances" instead of one that was the sum of the deposits. 

Essentially, with a deposit with each of 10K, Bob the balances map was: 

```
{
AztecAddress<Bob>: 10K,
AztecAddress<Bob>: 10K,
}
```
instead of `AztecAddress<Bob>: 20K`. Using the string makes them land at the same and have the expected map, kinda pain.